tests/e2e/tidbcluster/serial.go

```diff
@@ -589,6 +589,229 @@ var _ = ginkgo.Describe("[Serial]", func() {
+				// Dependent this PR: https://github.com/pingcap/tidb-operator/pull/3819
+				// AppendReleaseSuffix: true,
+			}
+			oa.DeployOperatorOrDie(ocfg2)
+			log.Logf("Finished deploying TiDB Operator 2 with --selector=version=new")
+
+			ginkgo.By("Check TidbCluster 2 is ready")
+			err = oa.WaitForTidbClusterReady(tc2, 6*time.Minute, 5*time.Second)
+			framework.ExpectNoError(err, "Expected TiDB cluster2 ready")
+			log.Logf("confirm TidbCluster 2 is ready")
+
+			ginkgo.By("Delete the default TiDB Operator")
+			oa.CleanOperatorOrDie(ocfg)
+			log.Logf("Finished delete default Operator")
+
+			ginkgo.By("Upgrade TiDB version of TidbCluster 2")
+			err = controller.GuaranteedUpdate(genericCli, tc2, func() error {
+				tc2.Spec.Version = utilimage.TiDBV4
+				return nil
+			})
+			framework.ExpectNoError(err, "failed to update TidbCluster 2 to upgrade tidb version to %v", utilimage.TiDBV4)
+			log.Logf("Finished upgrade TidbCluster 2")
+
+			ginkgo.By(fmt.Sprintf("wait for TidbCluster 2 tikv-0 pod upgrading to %q", utilimage.TiDBV4))
+			err = wait.Poll(5*time.Second, 10*time.Minute, func() (done bool, err error) {
+				tidbPod, err := c.CoreV1().Pods(ns).Get(fmt.Sprintf("%s-tikv-0", tc2.Name), metav1.GetOptions{})
+				if err != nil {
+					return false, nil
+				}
+				if tidbPod.Spec.Containers[0].Image != fmt.Sprintf("pingcap/tikv:%s", utilimage.TiDBV4) {
+					return false, nil
+				}
+				return true, nil
+			})
+			framework.ExpectNoError(err, "failed to upgrade TidbCluster 2 tikv-0 to %q", utilimage.TiDBV4)
+			log.Logf("Finished TidbCluster 2 been upgrade")
+
+			ginkgo.By("Deploy TiDB Operator 1 with --selector=version=old")
+			operatorVersion = "v1.1.11"
+			ocfg = &tests.OperatorConfig{
+				Namespace:       ns,
+				ReleaseName:     "operator",
+				Tag:             operatorVersion,
+				Image:           fmt.Sprintf("pingcap/tidb-operator:%s", operatorVersion),
+				ImagePullPolicy: v1.PullIfNotPresent,
+				Selector:        []string{"version=old"},
+			}
+			oa.DeployOperatorOrDie(ocfg)
+			log.Logf("Finished deploying TiDB Operator 1 with --selector=version=old")
+
+			ginkgo.By("Delete the TiDB Operator 2")
+			oa.CleanOperatorOrDie(ocfg2)
+			log.Logf("Finished delete Operator 2")
+
+			ginkgo.By("scale-out TiDB of TidbCluster 1, wait for 1min, check that scale up occurs")
+			err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+				tc.Spec.TiDB.Replicas = 2
+				return nil
+			})
+			framework.ExpectNoError(err, "failed to scale up TidbCluster 1")
+
+			err = wait.Poll(5*time.Second, 10*time.Minute, func() (done bool, err error) {
+				tidbStatefulSet, err := c.AppsV1().StatefulSets(ns).Get(fmt.Sprintf("%s-tidb", tc.Name), metav1.GetOptions{})
+				if err != nil {
+					return false, nil
+				}
+				if *tidbStatefulSet.Spec.Replicas != 2 {
+					return false, nil
+				}
+				return true, nil
+			})
+			framework.ExpectEqual(err, wait.ErrWaitTimeout, "expect tidbcluster 1 been scale out to 2")
```

Comment:
Should not expect `ErrWaitTimeout` here.

Reply:
fix